lib/CallStack.cpp

```diff
@@ -75,6 +75,16 @@ QoreHashNode* CallNode::getInfo() const {
          ph->setKeyValueIntern("type",  new QoreStringNode("new-thread"));
          break;
    }
+   if (pgm) {
+      ph->setKeyValueIntern("programid", pgm->getProgramId());
+      if (statement) {
+         unsigned long sid = pgm->getStatementId(statement);
+         if (sid) {
+            ph->setKeyValueIntern("statementid", sid);
+         }
+      }
+   }
+
    return h;
 }
 
```

Comment:
also this file has a commented out code below (cannot add the comment directly there), can you please resolve it?

Reply:
If you mean `void CallStack::substituteObjectIfEqual(QoreObject *o) {
` is not related to this commit at all.

Reply:
true, it's not, so I will open a new issue then

Reply:
https://github.com/qorelanguage/qore/issues/2971